infinite/index.ts

```diff
@@ -171,11 +174,11 @@ export const infinite = (<Data, Error>(useSWRNext: SWRHook) =>
             revalidateAll ||
             forceRevalidateAll ||
             isUndefined(pageData) ||
-            (revalidateFirstPage && !i && !isUndefined(dataRef.current)) ||
+            (revalidateFirstPage && !i && !isUndefined(cacheData)) ||
             shouldRevalidateOnMount ||
-            (originalData &&
-              !isUndefined(originalData[i]) &&
-              !config.compare(originalData[i], pageData))
+            (cacheData &&
+              !isUndefined(cacheData[i]) &&
+              !config.compare(cacheData[i], pageData))
```

Comment:
Q: Previously, `originalData` was set only when the revalidation is caused by `mutate` that has `data` and enable the `revalidate` option. Does this change any behavior?

Reply:
It would not change the behavior here. 

```set({ _i: [true] })``` will set `forceRevalidateAll` to `true`, then the expression result will be `true`. The value of `originalData(cacheData)` would be ignored here